client/src/app/components/analysis/steps/custom-rules.tsx

```
@@ -48,12 +50,59 @@ import { useFetchIdentities } from "@app/queries/identities";
+  isCustomRuleRequired: boolean;
+}): yup.SchemaOf<CustomRulesStepValues> => {
+  return yup.object({
+    rulesKind: yup.mixed<"manual" | "repository">().required(),
+
+    // manual tab fields
+    customRulesFiles: yup
+      .array()
+      .of(UploadFileSchema)
+      .when("rulesKind", {
+        is: "manual",
+        then: (schema) =>
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)